pkg/bench/ring_check.go

```diff
@@ -53,15 +54,25 @@ func NewRingChecker(id string, instanceName string, cfg RingCheckConfig, workloa
 		logger:   logger,
 		workload: workload,
 	}
-	cfg.MemberlistKV.MetricsRegisterer = prometheus.DefaultRegisterer
+	reg := prometheus.DefaultRegisterer
+	cfg.MemberlistKV.MetricsRegisterer = reg
 	cfg.MemberlistKV.Codecs = []codec.Codec{
 		ring.GetCodec(),
 	}
-	r.MemberlistKV = memberlist.NewKVInitService(&cfg.MemberlistKV, logger)
+
+	dnsProviderReg := prometheus.WrapRegistererWithPrefix(
+		"cortex_",
+		prometheus.WrapRegistererWith(
+			prometheus.Labels{"name": "memberlist"},
+			reg,
+		),
+	)
+	dnsProvider := dns.NewProvider(logger, dnsProviderReg, dns.GolangResolverType)
+	r.MemberlistKV = memberlist.NewKVInitService(&cfg.MemberlistKV, logger, dnsProvider, dnsProviderReg)
```

Comment:
Sorry about the confusion from before, but @pstibrany pointed out that we haven't historically prefixed memberlist metrics with `cortex_`, so it shouldn't be done here either most likely.
```suggestion
	r.MemberlistKV = memberlist.NewKVInitService(&cfg.MemberlistKV, logger, dnsProvider, reg)
```